### PR TITLE
[terra-data-grid] Resolved data grid focus management issues.

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The DataGrid was updated so that it can properly handle tab focus management when the next focusable element in the DOM is hidden.  The tab focus management was also updated to wrap when the DataGrid is the last focusable element in the DOM.

**Why it was changed:**
The change was made to resolve issues found during integration testing.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9686 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-data-grid/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Fixed
+  * Fixed tab focus management when next element is in a hidden container.
+  * Fixed tab focus maanagement to wrap to the first focusable element when the data grid is the last focusable element.
```
Comment (sycombs) [inline packages/terra-data-grid/CHANGELOG.md:7]: nit: typo
```suggestion
  * Fixed tab focus management to wrap to the first focusable element when the data grid is the last focusable element.
```

**Review by sycombs — APPROVED** [review]

**Review by eawww — COMMENTED** [review]
Issue still exists in cases where adjacent focusable element is contained in:
- `visibility: hidden`
- `inert`

File: packages/terra-data-grid/CHANGELOG.md
Comment (sdadn) [inline packages/terra-data-grid/CHANGELOG.md:1]: These entries will need to be moved to the unreleased section

File: packages/terra-framework-docs/CHANGELOG.md
Comment (sdadn) [inline packages/terra-framework-docs/CHANGELOG.md:1]: Same here

**Review by sdadn — APPROVED** [review]

**Comment by chrismichalewicz** [comment]
@cm9361 As @eawww  is OOO performed UX review and looked good. Update UX Reviewed label

**Pull request merged**